src/js/jsx/sections/layers/LayerFace.jsx

```diff
@@ -167,7 +167,14 @@ define(function (require, exports, module) {
                 }
             }
 
-            this.getFlux().actions.layers.select(this.props.document, this.props.layer, modifier);
+            // The clicked layer may an have out-of-date document models due to
+            // the aggressive SCU method in LayersPanel.
```

Comment:
ramble: seems like the most react-y way to do this would be for the method that calls the flux action be at the same component as the aggressive SCU, and would thus have ready access to current props/state.  That could be wrapped in a function and provided as a property, pushed in to the sub-component which may not get re-rendered very often.  Maybe that doesn't make sense here for some reason... maybe the aggressive SCU is *too high* up the chain.